### DISCUSSION TITLE
Set active layer state based on dynamic layer returns

### DISCUSSION THREAD

**Discussion by TangoYankee** [NYCPlanning/ae-zoning-map-poc#31] [General]
# Description

The content of the layers will be dynamic based on data from the API. The state of the layers needs to be able to dynamically expand and contract based on this data. 

## Possible approach
1. Use a set that holds a list of active layer ids. 
2. Check the visibility of the layer by checking whether the layer exists within the set of active layer ids
3. Update the visibility by adding or removing the layer id. 

## Things to ignore
The use of react context and react reducer are incidental. The problem of dynamically setting layer state exists regardless of whether that state is managed using context. 

## Code example
[Use array of active layers](https://github.com/NYCPlanning/ae-zoning-map-poc/commit/60ea33bb292ad997c7b3787b59ec3f18bd704028)

**Comment by dhochbaum-dcp** [comment]
Thank you.  For some reason my brain simply never managed to have the idea to store a list of ids rather than an object with a bunch of booleans.  That solves the problem of the dynamic state.  I'm going to close this out, we can re-open if there are any other items to discuss.
